### PR TITLE
Modify cn and memorystorge to be compatible with tae's behavior

### PR DESCRIPTION
**Pull request by nnsgmsone**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [X] Code Refactoring

## Which issue(s) this PR fixes:

issue #3931

## What this PR does / why we need it:

### CODE REVIEW

**Review by reusee — APPROVED** [review]

File: pkg/vm/engine/disttae/table.go [hunk 1]
```diff
@@ -136,18 +139,44 @@ func (tbl *table) getTableDef() *plan.TableDef {
...
+		viewDef.View = tbl.viewdef
+		defs = append(defs, viewDef)
+	}
 	for i, def := range tbl.defs {
-		switch attr := def.(type) {
-		case *engine.CommentDef:
-			defs = append(defs, tbl.defs[i])
-		case *engine.AttributeDef:
+		if attr, ok := def.(*engine.AttributeDef); ok {
 			if attr.Attr.Name != catalog.Row_ID {
```
Comment (fengttt) [inline pkg/vm/engine/disttae/table.go:162]: Test IsPhyAddr?

**Review by fengttt — APPROVED** [review]
I think this is OK -- but it is definitely not good code.   I still believe TableDef should only be used in show.    Resolve should just resolve over columns.

Please file an issue to cleanup this stuff.

**Review by ouyuanning — APPROVED** [review]

**Comment by nnsgmsone** [comment]
https://github.com/matrixorigin/matrixone/issues/5590

**Review by XuPeng-SH — APPROVED** [review]
Some follow-up work is required to get the commented out unit test to work

**Review by aunjgr — APPROVED** [review]

**Pull request merged**